executor/common_linux.h

```diff
@@ -5215,3 +5215,64 @@ static long syz_80211_join_ibss(volatile long a0, volatile long a1, volatile lon
 }
 
 #endif
+
+#if SYZ_EXECUTOR || __NR_syz_clone || __NR_syz_clone3
+#if SYZ_EXECUTOR
+// The slow down multiplier is already taken into account.
+#define USLEEP_FORKED_CHILD (3 * syscall_timeout_ms)
+#else
+// It's impossible to use replacements here, so we just hardcode it.
+#define USLEEP_FORKED_CHILD (3 * /*{{{BASE_CALL_TIMEOUT_MS}}}*/ *1000)
+#endif
+
+#ifndef CLONE_VM
+#define CLONE_VM 0x00000100
+#endif
+
+static long handle_clone_ret(long ret)
+{
+	if (ret != 0)
+		return ret;
+	// Exit if we're in the child process - not all kernels provide the proper means
+	// to prevent fork-bombs.
+	// But first sleep for some time. This will hopefully foster IPC fuzzing.
+	usleep(USLEEP_FORKED_CHILD);
+	// Note that exit_group is a bad choice here because if we created just a thread, then
+	// the whole process will be killed. A plain exit will work fine in any case.
+	syscall(__NR_exit, 0);
+	while (1) {
+	}
+}
+#endif
+
+#if SYZ_EXECUTOR || __NR_syz_clone
+// syz_clone is mostly needed on kernels which do not suport clone3.
+static long syz_clone(volatile long a0, volatile long a1, volatile long a2, volatile long a3, volatile long a4)
+{
+	// Clear the CLONE_VM flag. Otherwise it'll very likely corrupt syz-executor.
+	long ret = (long)syscall(__NR_clone, a0 & ~CLONE_VM, a1, a2, a3, a4);
```

Comment:
For clone `sp` must point to the _end_ of the stack (top word) (for arches where stack grows down, i.e. all arhes). We can't describe this using descriptions (?), but we can easily fix it up here if we also pass in size.

Reply:
Hmm, interesting. Added the extra stack len parameter.